tree.go

```diff
@@ -449,75 +451,39 @@ func (tree *Tree) newnode() (p *node) {
 	return &(tree.alloc[ln])
 }
 
-func loadip4(ipstr []byte) (uint32, error) {
-	var (
-		ip  uint32
-		oct uint32
-		b   byte
-		num byte
-	)
-
-	for _, b = range ipstr {
-		switch {
-		case b == '.':
-			num++
-			if 0xffffffff-ip < oct {
-				return 0, ErrBadIP
-			}
-			ip = ip<<8 + oct
-			oct = 0
-		case b >= '0' && b <= '9':
-			oct = oct*10 + uint32(b-'0')
-			if oct > 255 {
-				return 0, ErrBadIP
-			}
-		default:
-			return 0, ErrBadIP
-		}
-	}
-	if num != 3 {
-		return 0, ErrBadIP
-	}
-	if 0xffffffff-ip < oct {
-		return 0, ErrBadIP
-	}
-	return ip<<8 + oct, nil
-}
+func parseCIDR(cidr []byte) (net.IP, net.IPMask, error) {
```

Comment:
It's probably worth documenting here that you're checking for IPv4-mapped IPv6 addresses, and unmapping those back to IPv4 addresses